components/go/pkg/config/config.go

```diff
@@ -0,0 +1,27 @@
+package config
+
+type LicenseConfig struct {
+	ProductName      string
+	EntitlementID    string
+	LicenseServerURL string
+	ExecutableName   string
+}
+
+var cfg *LicenseConfig
+
+func NewConfig(c *LicenseConfig) {
+	cfg = c
+}
+
+func SetConfig(name, entitlementID, URL, executable string) {
```

Comment:
if multiple goroutines are accessing or modifying cfg, it could lead to race conditions.
Might be you have to use sync.Mutex. Not so important

Reply:
This is a one-time config set, and we are also not using the go routines. I'll add a FIXME tag to address it later.